packages/runtime/test/execute/plan.test.ts

```diff
@@ -1204,3 +1206,39 @@ test('Plans log step names for each job start and end', async (t) => {
   const end = logger._find('success', /do-the-thing completed in/i);
   t.regex(end!.message as string, /do-the-thing completed in \d+ms/);
 });
+
+test.serial(
+  'global functions should be scoped per step or job code',
+  async (t) => {
+    const functions = `
```

Comment:
This code is quite complex :thinking: 

The exports here don't have to be operations or anything

Could we do more like:
```
let x = 10;
export const setX = (value) => { x = value }
export const getX = () => x
```
With step 1 like:
```
export default [
   s => {
      setX(20)
      return { x1: getX() }
   }
]
```
And step 2 like:
```
export default [
   s => {
      return { x2: getX() }
   }
]
```

Then x1 should be 20 and x2 should be 10

Reply:
Yeah makes sense. this simplifies it.

Reply:
but then we also need to be sure that it's shared between operations. probably a different test for that. thanks